app/src/main/java/de/hhn/aib/labsw/blackmirror/controller/PageController.java

```diff
@@ -194,4 +202,74 @@ private void autoChangePageByTime() {
+        }
+
+        try {
+            LayoutData data = nodeToObject(object, LayoutData.class);
+
+            // iterate the pages
+            for (int i = 0; i < data.pages().size(); i++) {
+
+                //create new page
+                ArrayList<AbstractWidgetController> page = new ArrayList();
+
+                // process all widgets for this page
+                if (data.pages().get(i).widgets().size() > 0) {
+                    for (int j = 0; j < data.pages().get(i).widgets().size(); j++) {
+                        WidgetData widget = data.pages().get(i).widgets().get(j);
+                        switch (widget.type()) {
+                            case CALENDAR -> {
+                                CalendarWidgetController calendarWidgetController = new CalendarWidgetController();
+                                calendarWidgetController.getWidget().setPosition(widget.x(), widget.y());
+                                page.add(calendarWidgetController);
+                            }
+                            case CLOCK -> {
+                                // todo : FaceType handling
+                                ClockWidgetController clockWidgetController = new ClockWidgetController(ClockFaceType.ANALOG);
+                                clockWidgetController.getWidget().setPosition(widget.x(), widget.y());
+                                page.add(clockWidgetController);
+                            }
+                            case MAIL -> {
+                                EmailNotificationController emailNotificationController = new EmailNotificationController();
+                                emailNotificationController.getWidget().setPosition(widget.x(), widget.y());
+                                page.add(emailNotificationController);
+                            }
+                            case REMINDER -> {
+                                TodosWidgetController todosWidgetController = new TodosWidgetController();
+                                todosWidgetController.getWidget().setPosition(widget.x(), widget.y());
+                                page.add(todosWidgetController);
+                            }
+                            case WEATHER -> {
+                                WeatherWidgetController weatherWidgetController = new WeatherWidgetController();
+                                weatherWidgetController.getWidget().setPosition(widget.x(), widget.y());
+                                page.add(weatherWidgetController);
+                            }
+                        }
+                    }
+                    addPageAtIndex(i, page);
+                }
+            }
+            pageIndex = data.currentPageIndex();
+            getCurrentPage().setWidgetsVisible();
+        } catch (/*JsonProcessingException*/ Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public AbstractWidget getWidget() {
+        return null;
+    }
```

Comment:
This method can be removed when the suggested change to the inheritance hierarchy has been made

Reply:
Done.